lib/rspec/openapi/default_schema.rb

```diff
@@ -1,13 +1,22 @@
 class << RSpec::OpenAPI::DefaultSchema = Object.new
   def build(title)
-    {
+    spec = {
       openapi: '3.0.3',
       info: {
         title: title,
         version: RSpec::OpenAPI.application_version,
       },
       servers: RSpec::OpenAPI.servers,
       paths: {},
-    }.freeze
+    }
+
+    if RSpec::OpenAPI.security_schemes.present?
+      spec[:components] = {
+        securitySchemes: RSpec::OpenAPI.security_schemes
```

Comment:
rspec-openapi does not touch "unknown" field added by users manually.
In other words, users can freely define securitySchemas in component block.
So I am inclined to reject this feature to reduce maintenance cost.

Could you elaborate why this feature is useful to users?

Reply:
It's part of the OpenAPI spec, static-ish (unlike components.schemas which, it should neither change often, nor be dynamically defined).

Users that like their open api spec to document which authentication mechanism their endpoints use would benefit from this change by being able to define it in their app's code instead of having it to author the openapi file by hand

Reply:
Understood.